sql/core/src/main/scala/org/apache/spark/sql/package.scala

```diff
@@ -98,7 +98,10 @@ package object sql {
       f
     } else {
       val st = Thread.currentThread().getStackTrace
-      var i = 3
+      var i = 0
+      // Find the beginning of Spark code traces
+      while (i < st.length && !sparkCode(st(i))) i += 1
+      // Stop at the end of the first Spark code traces
       while (i < st.length && sparkCode(st(i))) i += 1
       val origin =
         Origin(stackTrace = Some(Thread.currentThread().getStackTrace.slice(i - 1, i + 1)))
```

Comment:
Just in case, call to `slice` is safe since it checks borders, see:
```scala
  def slice(from: Int, until: Int): Array[A] = {
    import java.util.Arrays.copyOfRange
    val lo = max(from, 0)
    val hi = min(until, xs.length)
```